docs/Authenticating-and-Authorizing-Users-via-keycloak.md

```diff
@@ -357,3 +357,32 @@ log4j.logger.org.springframework.security=DEBUG
 ```
 
 Then, rebuild the WAR, redeploy, and try to authenticate again.  Your log file will then include hundreds of SAML-specific messages, even the full XML of each SAML message, and this should help you debug the error.
+
+#### Determining jwtRolesPath for OAuth2 Token
+If the generated token returns a 4XX error when used, confirm that the `jwtRolesPath` property matches the roles path. 
```

Comment:
The trigger for looking at this section should not be a 4xx error. I propose to remove this line and replace it for something like this:

> By default user-roles are extracted from path `resource_access::cbioportal::roles` in the JWT json. When the configuration of roles at the realm and client level in a Keycloak instance differ from that described in this manual, configure the path can be specified using the `dat.oauth2.jwtRolesPath` property in the _portal.properties_ file.